pkg/snapshot/snapshotter/snapshotter.go

```diff
@@ -504,12 +523,14 @@ func (ssr *Snapshotter) CollectEventsSincePrevSnapshot(stopCh <-chan struct{}) (
 	metrics.SnapshotRequired.With(prometheus.Labels{metrics.LabelKind: brtypes.SnapshotKindFull}).Set(1)
 
 	for {
+		leaseUpdateCtx, leaseUpdateCancel := context.WithCancel(context.TODO())
```

Comment:
I suggest to drop the creation of this new context and instead change the function:
```
func (ssr *Snapshotter) CollectEventsSincePrevSnapshot(stopCh <-chan struct{}) -> func (ssr *Snapshotter) CollectEventsSincePrevSnapshot(ctx context.Context)
```

It should be safe to pass the `leCtx` to this function then.